src/Config.js

```diff
@@ -63,35 +68,90 @@ export class Config {
     return serverConfiguration;
   }
 
-  static validateOptions({
-    publicServerURL,
-    revokeSessionOnPasswordReset,
-    expireInactiveSessions,
-    sessionLength,
-    defaultLimit,
-    maxLimit,
-    accountLockout,
-    passwordPolicy,
-    masterKeyIps,
-    masterKey,
-    maintenanceKey,
-    maintenanceKeyIps,
-    readOnlyMasterKey,
-    allowHeaders,
-    idempotencyOptions,
-    fileUpload,
-    pages,
-    security,
-    enforcePrivateUsers,
-    schema,
-    requestKeywordDenylist,
-    allowExpiredAuthDataToken,
-    logLevels,
-    rateLimit,
-    databaseOptions,
-    extendSessionOnUse,
-    allowClientClassCreation,
-  }) {
+  static validateConfigKeyNames(actual, given, parent) {
+    const internalConfigVars = [
+      'level',
```

Comment:
If a config key is missing, then it needs to be added to the Definitions file, not to a separate list that needs to be maintained. The Definitions file should be the only source of truth. Only the actual "internal" keys like `_mount` which is not an exposed key that developer can actually set should be added to an exclusion list, so that it's ignored by the validation.

Reply:
Most of these keys were pushed by the internal code rather than the user, so I didn't it looked unnecessary to check them since these keys won't ever give errors.

Reply:
Oh I see, never mind then, all good.

Reply:
But you were right about the maintenance part. As code changes and people keep adding keys to config this will need to be updated otherwise a lot of console logs might cause inconvenience with development and testing.